recompile_local_debian_package.sh

```diff
@@ -23,73 +23,246 @@ Written by Stéphane Gourichon <stephane_dpkg@gourichon.org>
+                        "$PKGDIR/debian/control" \
+                        -s "${BECOMEROOT:-}" ;
+        )
+        echo -e "* dpkg-checkbuilddeps\tPASSED"
+    fi
+
+    if [[ "${1:-}" == "deps_only" ]]
+#    if [[ "${local_install_deps_only:-}" == "true" ]]
+    then
+        echo "Install-only pass, returning now."
+        return 0
+    fi
+
+    CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD) && echo "* Current git branch $CURRENT_BRANCH"
+
+    NAMEFORTAR="$( head -n 1 debian/changelog | sed -n 's/^\([^ ]*\)* (\([^)]*\)).*$/\1_\2/p' )"
+    DIRNAMEFORDEB=${NAMEFORTAR//_/-}
+
+    if [[ -z "$NAMEFORTAR" ]]
+    then
+        echo >&2 "Cannot figure out tar archive name from first line of debian/changelog. Aborting"
+        head -n 1 debian/changelog
+        exit 1
+    fi
+
+    if output=$(git status --porcelain) && [ -z "$output" ]; then
+        echo "Working directory clean"
+    else
+        echo >&2 "WARNING: uncommitted changes. Consider aborting."
+        git status
+        echo >&2 "WARNING: uncommitted changes. Consider aborting."
+        echo >&2 "Waiting for 10 second."
+        #sleep 10
+    fi
+
+    GITREV=$( git describe )
+
+    cd "$TMPDIR"
+    git clone "$PKGDIR" "${DIRNAMEFORDEB}"
+
+    tar zcf "${NAMEFORTAR}.orig.tar.gz" "${DIRNAMEFORDEB}"
+    cd "${DIRNAMEFORDEB}"
 
-GITREV=$( git describe )
+    dpkg-checkbuilddeps
+    debuild -us -uc
 
-cd "$TMPDIR"
-git clone "$PKGDIR" "${DIRNAMEFORDEB}"
+    . /etc/os-release ; DISTRO_ID="${ID}-${VERSION_ID}"
 
-tar zcvf ${NAMEFORTAR}.orig.tar.gz "${DIRNAMEFORDEB}"
-cd "${DIRNAMEFORDEB}"
+    OUTDIR="$PKGDIR/../compiled_packages/${DISTRO_ID}/${NAMEFORTAR}" #_$( date +%Yy%mm%dd_%Hh%Mm%Ss )"
 
-dpkg-checkbuilddeps
-debuild -us -uc
+    mkdir -p "$OUTDIR"
 
-. /etc/os-release ; DISTRO_ID="${ID}-${VERSION_ID}"
+    echo "generated from git commit $GITREV" >"${OUTDIR}/${NAMEFORTAR}.gitversion"
 
-OUTDIR="$PKGDIR/../compiled_packages/${DISTRO_ID}/${NAMEFORTAR}" #_$( date +%Yy%mm%dd_%Hh%Mm%Ss )"
+    cd ..
 
-mkdir -p "$OUTDIR"
+    cp -v "${NAMEFORTAR}"?* "$OUTDIR"
 
-echo "generated from git commit $GITREV" >"${OUTDIR}/${NAMEFORTAR}.gitversion"
+    echo
+    echo ================================================================
+    echo "Artifacts available in $OUTDIR:"
+    echo "OUTDIR=$OUTDIR"
+    echo ================================================================
 
-cd ..
+    cd "$OUTDIR"
+
+    ls -al
+}
+
+function build_in_docker()
+{
+    OSIMAGE="$1"
+
+    # By default, files generated in Docker have same owner as user inside docker.
+    # So, let's make sure there is a matching user inside the docker container.
+
+    docker run -it -v "$PKGDIR/..":/up "$OSIMAGE" \
+           bash -c "
+cd /up/*/debian/source ; cd ../..
+bash recompile_local_debian_package.sh --install-deps-only
+USER=$(id -un) ; userdel \"\$USER\"
+GID=$(id -g) ; groupadd -f -g \$GID $(id -gn)
+useradd -u $(id -u) --gid stephane -l -m \"\$USER\" -s /bin/bash
```

Comment:
You may be `stephane`, but I'm `ssokolow` and have no such user on my system. It looks like you're either missing a username lookup here or should use a username that makes it more clear that all that matters is matching the external UID.

Reply:
Oops! Of course this is unacceptable for published software. Probably a local expedient that I forgot to refine.